src/anyio/_core/_synchronization.py

```diff
@@ -101,6 +102,32 @@ def statistics(self) -> EventStatistics:
         raise NotImplementedError
 
 
+class EventAdapter(Event):
+    _internal_event: Event | None = None
+
+    def __new__(cls) -> EventAdapter:
+        return object.__new__(cls)
+
+    @property
+    def _event(self) -> Event:
+        if self._internal_event is None:
+            self._internal_event = get_async_backend().create_event()
+
+        return self._internal_event
+
+    def set(self) -> None:
+        self._event.set()
+
+    def is_set(self) -> bool:
+        return self._internal_event is not None and self._internal_event.is_set()
```

Comment:
I understand that if the event is not created then it cannot be set, but should `is_set` actually create the event?
```suggestion
        return self._event.is_set()
```
`statistics` also doesn't need the event to be created (it could return empty statistics if it's not), yet it creates it.

Reply:
Added.